inference-engine/include/ie_common.h

```diff
@@ -274,73 +281,210 @@ struct QueryNetworkResult {
     ResponseDesc resp;
 };
 
-/** @brief This class represents StatusCode::GENERIC_ERROR exception */
-class GeneralError : public std::logic_error {
-    using std::logic_error::logic_error;
+namespace details {
+struct INFERENCE_ENGINE_DEPRECATED("Use InferRequest::Exception")
+INFERENCE_ENGINE_API_CLASS(InferenceEngineException) : public std::runtime_error {
+    using std::runtime_error::runtime_error;
+    bool hasStatus() const {return true;}
+    StatusCode getStatus() const;
 };
+}  // namespace details
 
-/** @brief This class represents StatusCode::NOT_IMPLEMENTED exception */
-class NotImplemented : public std::logic_error {
-    using std::logic_error::logic_error;
+/**
+ * @brief Base Inference Engine exception class
+ */
+IE_SUPPRESS_DEPRECATED_START
+struct INFERENCE_ENGINE_API_CLASS(Exception) : public details::InferenceEngineException {
+    using InferenceEngineException::InferenceEngineException;
 };
+IE_SUPPRESS_DEPRECATED_END
+
+/// @cond
+namespace details {
+    template<typename ExceptionType> struct ExceptionTraits;
+}
+
+#define INFERENCE_ENGINE_DECLARE_EXCEPTION(ExceptionType, statusCode)                           \
+struct INFERENCE_ENGINE_API_CLASS(ExceptionType) final : public InferenceEngine::Exception {    \
+    using Exception::Exception;                                                                 \
+};                                                                                              \
+namespace details {                                                                             \
+template<> struct ExceptionTraits<ExceptionType> {                                              \
+    constexpr static const StatusCode _statusCode = statusCode;                                 \
+    static const char* _string() {return "[ " #statusCode " ]";}                                \
```

Comment:
looks like old behavior was `"[" #statusCode "]"`

BTW, looks like `_statusCode` is not used